src/main/java/org/folio/circulation/services/FeeFineFacade.java

```diff
@@ -147,9 +150,22 @@ public FeeFineFacade(Clients clients) {
       lostItemRefundProcessor.onHasRemainingAmount(context);
     }
 
+    // Order actions historically with 1ms step
+    // Because they are created very quickly
+    // So the ISO date time is the same for all actions
+    populateActionDateHistorically(context);
+
     return feeFineActionRepository.createAll(context.getActions());
   }
 
+  private void populateActionDateHistorically(AccountRefundContext context) {
+    final DateTime now = getClockManager().getDateTime();
+
+    range(0, context.getActions().size())
+      .forEach(index -> context.getActions().get(index)
+        .updateActionDate(now.plusMillis(index)));
```

Comment:
There is a `dateAction` property for each fee/fine action which means date and time of the transaction. UI uses this field to order actions. 

Mod-circulation constructs refund/cancellation fee/fine actions very quickly (within nano seconds) and because of this `dateAction` property ISO string is the same for all of the created actions. This causes UI to show the actions in wrong order: e.g. remaining amount for action N is more then for action N + 1. 

The proper order is the order in which we create actions:
1. Cancellation;
2 Payment credit/refund;
3 Transfer credit/refund.

Reply:
This is now moved to the `org.folio.circulation.services.feefine.FeeRefundProcessor`.

Reply:
Does the code manipulate part of the date-time to add fake precision to force the ordering?

Reply:
DateAction = now() + indexOfAction ms

Reply:
Ok, thanks. It's a shame that a more precise date time couldn't be generated directly. Thank you for improving the clarity of the code for this.